examples/viewer.js

```diff
@@ -36,16 +38,239 @@ let droppedFramePerArray = [];
+            name: 'signaling-master-get-signaling-channel-endpoint',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to getSignalingChannelEndpoint on the master',
+            color: '#EF9A9A',
+        },
+        iceServerConfig: {
+            name: 'signaling-master-get-ice-server-config',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the API call to getIceServerConfig on the master',
+            color: '#EF9A9A',
+        },
+        getToken: {
+            name: 'signaling-master-get-token',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the getToken call on the master',
+            color: '#EF9A9A',
+        },
+        createChannel: {
+            name: 'signaling-master-create-channel',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken createChannel API call on the master',
+            color: '#EF9A9A',
+        },
+        connectAsMaster: {
+            name: 'signaling-master-connect',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken for the signaling connect on the master',
+            color: '#EF9A9A',
+        },
+        iceGathering: {
+            name: 'ice-gathering-master',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to gather all ice candidates on the master',
+            color: '#90CAF9',
+        }, 
+        peerConnection: {
+            name: 'pc-establishment-master',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to establish the peer connection on the master',
+            color: '#2196F3',
+        },
+        dataChannel: {
+            name: 'datachannel-master',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time taken to send a message to the viewer and receive a response back',
+            color: '#4CAF50',
+        },
+        ttffAfterPc: {
+            name: 'ttff-after-pc-master',
+            startTime: '',
+            endTime: '',
+            tooltip: 'Time to first frame after the master\'s peer connection has been established',
+            color: '#2196F3',
+        }
+    }
+};
+
+let dataChannelLatencyCalcMessage = {
+    content: 'Opened data channel by viewer',
+    timestamp1: '',
+    timestamp2: '',
+    timestamp3: '',
+    timestamp4: '',
+    timestamp5: ''
+}
 
-async function startViewer(localView, remoteView, formValues, onStatsReport, onRemoteDataMessage) {
+async function startViewer(localView, remoteView, formValues, onStatsReport, remoteMessage) {
     try {
         console.log('[VIEWER] Client id is:', formValues.clientId);
 
         viewer.localView = localView;
         viewer.remoteView = remoteView;
 
+        viewer.remoteView.addEventListener('loadeddata', () => {
+            metrics.viewer.video.endTime = Date.now();
```

Comment:
Is the intent here to record time to first decoded frame? If so, the naming seems misleading. Can we be specific with the naming here?

Also, this event should be listened to only if DQP is enabled or timeline display is enabled, alright if we dont want to have that check, but I dont think we need to do this needlessly either.

Reply:
Changed the name. Introduced a new flag which is checked when this feature is to be used. Added a screenshot in the description